openshift/templates/backup/backup-build.yaml

```diff
@@ -0,0 +1,72 @@
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: "${NAME}-build-template"
+  creationTimestamp:
+objects:
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: "${NAME}"
+  - kind: BuildConfig
+    apiVersion: v1
+    metadata:
+      name: "${NAME}"
+      labels:
+        app: "${NAME}"
+    spec:
+      triggers:
+        - type: ImageChange
+        - type: ConfigChange
+      runPolicy: Serial
+      source:
+        type: Git
+        git:
+          uri: "${GIT_REPO_URL}"
+          ref: "${GIT_REF}"
+        contextDir: "${SOURCE_CONTEXT_DIR}"
+      strategy:
+        type: Docker
+        dockerStrategy:
+          dockerfilePath: "${DOCKER_FILE_PATH}"
+      output:
+        to:
+          kind: ImageStreamTag
+          name: "${NAME}:${OUTPUT_IMAGE_TAG}"
+parameters:
+  - name: NAME
+    displayName: Name
+    description:
+      The name assigned to all of the resources.  Use 'backup-{database name}'
+      depending on your database provider
+    required: true
+    value: backup-mssql
+  - name: GIT_REPO_URL
+    displayName: Git Repo URL
+    description: The URL to your GIT repo.
+    required: true
+    value: https://github.com/devinleighsmith/backup-container.git
+  - name: GIT_REF
+    displayName: Git Reference
+    description: The git reference or branch.
+    required: true
+    value: mssql
+  - name: SOURCE_CONTEXT_DIR
+    displayName: Source Context Directory
+    description: The source context directory.
+    required: false
+    value: "/docker"
+  - name: DOCKER_FILE_PATH
+    displayName: Docker File
+    description:
+      The path and file of the docker file defining the build.  Choose either
+      'Dockerfile' for Postgres builds or 'Dockerfile_Mongo' for MongoDB builds or 'Dockerfile_MSSQL'
+      for MSSQL builds.
+    required: false
+    value: Dockerfile_MSSQL
```

Comment:
I don't think I see these dockerfiles anywhere?  Should they be part of the PR?

Reply:
If I understood correctly, they are coming from a separate "base" repo that has all the logic related to the `backup-container`

https://github.com/BCDevOps/backup-container/blob/master/docker/Dockerfile

Reply:
Yes, I think it makes sense to include a minimal amount of files from the backup container repo for tracking purposes - but in general we should use the base repo because that is where new features fixes and updates will show up.